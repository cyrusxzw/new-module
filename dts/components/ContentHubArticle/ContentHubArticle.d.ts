export function ContentHubArticle({
  articleRef,
  category,
  className,
  dataTestRef,
  horizontalThumbnail,
  id,
  isHorizontal,
  isInFirstGroup,
  isMenuItem,
  isReadMore,
  longTitle,
  onClick,
  readingTime,
  uri,
  verticalThumbnail,
}: {
  articleRef: any;
  category: any;
  className: any;
  dataTestRef: any;
  horizontalThumbnail: any;
  id: any;
  isHorizontal: any;
  isInFirstGroup: any;
  isMenuItem: any;
  isReadMore: any;
  longTitle: any;
  onClick: any;
  readingTime: any;
  uri: any;
  verticalThumbnail: any;
}): JSX.Element;
export namespace ContentHubArticle {
  namespace propTypes {
    const articleRef: PropTypes.Requireable<object>;
    const category: PropTypes.Requireable<string>;
    const className: PropTypes.Requireable<string>;
    const dataTestRef: PropTypes.Validator<string>;
    const horizontalThumbnail: PropTypes.Requireable<object>;
    const id: PropTypes.Requireable<string>;
    const isHorizontal: PropTypes.Requireable<boolean>;
    const isInFirstGroup: PropTypes.Requireable<boolean>;
    const isMenuItem: PropTypes.Requireable<boolean>;
    const isReadMore: PropTypes.Requireable<boolean>;
    const longTitle: PropTypes.Requireable<string>;
    const onClick: PropTypes.Requireable<(...args: any[]) => any>;
    const readingTime: PropTypes.Requireable<string>;
    const uri: PropTypes.Requireable<string>;
    const verticalThumbnail: PropTypes.Requireable<object>;
  }
  namespace defaultProps {
    const articleRef_1: any;
    export { articleRef_1 as articleRef };
    const category_1: any;
    export { category_1 as category };
    const className_1: any;
    export { className_1 as className };
    const dataTestRef_1: any;
    export { dataTestRef_1 as dataTestRef };
    const horizontalThumbnail_1: any;
    export { horizontalThumbnail_1 as horizontalThumbnail };
    const id_1: any;
    export { id_1 as id };
    const isHorizontal_1: any;
    export { isHorizontal_1 as isHorizontal };
    const isInFirstGroup_1: boolean;
    export { isInFirstGroup_1 as isInFirstGroup };
    const isMenuItem_1: boolean;
    export { isMenuItem_1 as isMenuItem };
    const isReadMore_1: boolean;
    export { isReadMore_1 as isReadMore };
    const longTitle_1: any;
    export { longTitle_1 as longTitle };
    const onClick_1: any;
    export { onClick_1 as onClick };
    const readingTime_1: any;
    export { readingTime_1 as readingTime };
    const uri_1: any;
    export { uri_1 as uri };
    const verticalThumbnail_1: any;
    export { verticalThumbnail_1 as verticalThumbnail };
  }
}
import PropTypes from 'prop-types';
