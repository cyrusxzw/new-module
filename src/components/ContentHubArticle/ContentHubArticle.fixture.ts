const ContentHubArticleFixture = {
  id: 'test_id',
  category: 'Product',
  dataTestRef: 'test_id',
  longTitle:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse et ornare mauris, id dictum ante.',
  verticalThumbnail: {
    altText: 'vertical_hub',
    sizes: {
      large: './assets/images/ContentHubArticle/vertical_hub.jpg',
      medium: './assets/images/ContentHubArticle/vertical_hub.jpg',
      small: './assets/images/ContentHubArticle/vertical_hub.jpg',
    },
  },
  horizontalThumbnail: {
    altText: 'horizontal_hub',
    sizes: {
      large: './assets/images/ContentHubArticle/horizontal_hub.jpg',
      medium: './assets/images/ContentHubArticle/horizontal_hub.jpg',
      small: './assets/images/ContentHubArticle/horizontal_hub.jpg',
    },
  },
  uri: 'https://www.aesop.com/au/r/careers',
  readingTime: '5 minute read',
  onClick: () => {
    console.log('click tracking: test_id'); // eslint-disable-line no-console
  },
};

export { ContentHubArticleFixture };
