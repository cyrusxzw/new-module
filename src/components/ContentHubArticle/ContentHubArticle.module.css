@import 'styles/index';

.base {
  display: block;

  h1 {
    margin: 0;
    font-weight: normal;
  }

  @media (--viewport-xs-sm-only) {
    padding-bottom: 50px;
  }

  @media (--viewport-md-only) {
    &.readMoreArticle {
      padding-right: 0;
      padding-left: 0;
    }
  }

  @media (--viewport-lg) {
    &.readMoreArticle {
      padding-right: 20px;
      padding-left: 20px;
    }
  }
}

.category {
  padding-bottom: 14px;
  border-bottom: 1px solid var(--color-grey-rgba-20);
  margin-bottom: 9px;
  font-family: var(--font-family-sans-serif-medium);
  font-size: rem(14px);
  font-weight: medium;
  letter-spacing: 0;
  line-height: 1.7;

  &.readMoreCategory {
    margin-top: 20px;
    margin-bottom: 14px;
    line-height: 1.5;
  }

  &::after {
    display: inline-block;
    padding: 0 10px;
    border-bottom-color: var(--color-dark-copy);
    content: '\2022';
    font-size: rem(14px);
    line-height: 1.7;
  }
}

.title span {
  @extend %zaph-serif-typography;

  padding-bottom: 1.5px;
  border-bottom: 1px solid transparent;
  color: var(--color-grey-20);
  font-size: rem(30px);
  letter-spacing: 0;
  line-height: 1.2;
  transition: all 500ms var(--easing-ease-in-cubic);

  @media (--viewport-xs-sm-only) {
    font-size: rem(20px);
  }

  @media (--viewport-md-only) {
    font-size: rem(26px);
  }
}

.readMoreTitle span {
  font-size: rem(24px);
  line-height: 1.25;
}

.title:hover span {
  border-bottom: 1px solid var(--color-secondary);
}

.titleActive {
  border-bottom: 1px solid var(--color-secondary) !important;
  transition: all 500ms var(--easing-ease-in-cubic) !important;
}

.image:hover img {
  filter: grayscale(100%);
  transition: all 500ms var(--easing-ease-in-cubic);
}

.imageActive {
  filter: grayscale(100%);
  transition: all 500ms var(--easing-ease-in-cubic);
}

.imageUnactive {
  filter: grayscale(0);
  transition: all 500ms var(--easing-ease-in-cubic);
}

.note {
  margin: 15px 0;
  color: var(--color-grey-30);
  font-family: var(--font-family-sans-serif-medium);
  font-size: rem(16px);
  letter-spacing: 0;
  line-height: 1.5;

  &.readMoreNote {
    @media (--viewport-xs-sm-only) {
      font-size: rem(14px);
      line-height: 1.7;
    }
  }
}

.nonMobile {
  display: none;

  @media (--viewport-md) {
    display: block;
  }
}

.mobile {
  @media (--viewport-xs-sm-only) {
    display: block;
  }

  @media (--viewport-md) {
    display: none;
  }
}
