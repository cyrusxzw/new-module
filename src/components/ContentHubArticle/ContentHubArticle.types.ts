import type { MutableRefObject } from 'react';
import type { ComponentWithoutChildren } from '~/types';
import type { ImageProps } from '~/components/Image/Image.types';

type Image = Pick<ImageProps, 'altText' | 'sizes'>;

type Article = {
  articleRef?: MutableRefObject<HTMLDivElement>;
  category?: string;
  horizontalThumbnail?: Image;
  id?: string;
  longTitle?: string;
  onClick?: () => void;
  readingTime?: string;
  uri?: string;
  verticalThumbnail?: Image;
};

type ContentHubArticleProps = Article & {
  className?: string;
  dataTestRef: string;
  isHorizontal?: boolean;
  isInFirstGroup?: boolean;
  isMenuItem?: boolean;
  isReadMore?: boolean;
};

type ContentHubArticleType = ComponentWithoutChildren<ContentHubArticleProps>;

export type { ContentHubArticleProps, ContentHubArticleType, Article };
