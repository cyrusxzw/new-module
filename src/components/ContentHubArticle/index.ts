import { ContentHubArticle } from './ContentHubArticle';

export { ContentHubArticle };
