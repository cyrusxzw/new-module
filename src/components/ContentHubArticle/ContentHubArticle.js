import React, { useRef } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import { useOnScreen } from '~/customHooks';
import { Image } from '~/components/Image';
import { Hyperlink } from '~/components/Hyperlink';
import { Transition } from '~/components/Transition';
import styles from './ContentHubArticle.module.css';

const ContentHubArticle = ({
  articleRef,
  category,
  className,
  dataTestRef,
  horizontalThumbnail,
  id,
  isHorizontal, // use horizontal or vertical thumbnail
  isInFirstGroup,
  isMenuItem, // if it's displayed in Read Menu Navigation
  isReadMore, // if it's displayed in ReadMore section
  longTitle,
  onClick,
  readingTime,
  uri,
  verticalThumbnail,
}) => {
  const defaultRef = useRef(null);
  const ref = articleRef || defaultRef;
  const isOnScreen = useOnScreen(ref, 0.5);
  const currentImage = isHorizontal ? horizontalThumbnail : verticalThumbnail;
  const classSet = cx(styles.base, className, {
    [styles.readMoreArticle]: !!isReadMore,
  });
  const categoryClassSet = cx(styles.category, {
    [styles.readMoreCategory]: !!isReadMore,
  });
  const imageClassSet = cx(styles.image);
  const readMoreThumbnailRef = useRef(null);
  const nonMobileImageClassSet = cx(styles.image, styles.nonMobile);
  const nonMobileThumbnailRef = useRef(null);
  const noteClassSet = cx(styles.note, { [styles.readMoreNote]: !!isReadMore });
  const mobileImageClassSet = cx(styles.image, styles.mobile);
  const mobileThumbnailRef = useRef(null);
  const titleClassSet = cx(styles.title, {
    [styles.readMoreTitle]: !!isReadMore,
  });
  const titleRef = useRef(null);

  const imageMouseEnter = () => {
    const titleSpan = titleRef.current
      .getElementsByClassName(titleClassSet)[0]
      .getElementsByTagName('span')[0];
    titleSpan.className = cx(styles.title, styles.titleActive);
  };

  const imageMouseLeave = () => {
    const titleSpan = titleRef.current
      .getElementsByClassName(titleClassSet)[0]
      .getElementsByTagName('span')[0];
    titleSpan.className = '';
  };

  const titleMouseEnter = () => {
    [readMoreThumbnailRef, nonMobileThumbnailRef, mobileThumbnailRef].forEach(
      (imageRef) => {
        if (imageRef.current)
          imageRef.current.className = cx(styles.imageActive);
      },
    );
  };

  const titleMouseLeave = () => {
    [readMoreThumbnailRef, nonMobileThumbnailRef, mobileThumbnailRef].forEach(
      (imageRef) => {
        if (imageRef.current)
          imageRef.current.className = cx(styles.imageUnactive);
      },
    );
  };

  return (
    <Transition isActive={isInFirstGroup || isOnScreen} type="fade">
      <div className={classSet} id={id} ref={ref}>
        {isReadMore && !isMenuItem && (
          <Hyperlink
            className={imageClassSet}
            dataTestRef={`${dataTestRef}_READMORE_THUMBNAIL`}
            onClick={onClick}
            title={longTitle}
            url={uri}
          >
            <div
              onMouseEnter={() => imageMouseEnter()}
              onMouseLeave={() => imageMouseLeave()}
            >
              <Image
                altText={currentImage.altText}
                ref={readMoreThumbnailRef}
                sizes={currentImage.sizes}
              />
            </div>
          </Hyperlink>
        )}
        {!isMenuItem && <div className={categoryClassSet}>{category}</div>}
        <div
          onMouseEnter={() => titleMouseEnter()}
          onMouseLeave={() => titleMouseLeave()}
          ref={titleRef}
        >
          <Hyperlink
            className={titleClassSet}
            dataTestRef={`${dataTestRef}_TITLE`}
            onClick={onClick}
            title={longTitle}
            url={uri}
          >
            <h1>
              <span>{longTitle}</span>
            </h1>
          </Hyperlink>
        </div>
        {readingTime && (
          <div className={noteClassSet}>
            <span>{readingTime}</span>
          </div>
        )}

        {!isReadMore && !isMenuItem && (
          <Hyperlink
            className={nonMobileImageClassSet}
            dataTestRef={`${dataTestRef}_NON_MOBILE_THUMBNAIL`}
            onClick={onClick}
            title={longTitle}
            url={uri}
          >
            <div
              onMouseEnter={() => imageMouseEnter()}
              onMouseLeave={() => imageMouseLeave()}
            >
              <Image
                altText={currentImage.altText}
                className={styles.image}
                ref={nonMobileThumbnailRef}
                sizes={currentImage.sizes}
              />
            </div>
          </Hyperlink>
        )}
        {!isReadMore && !isMenuItem && (
          <Hyperlink
            className={mobileImageClassSet}
            dataTestRef={`${dataTestRef}_MOBILE_THUMBNAIL`}
            onClick={onClick}
            title={longTitle}
            url={uri}
          >
            <div
              onMouseEnter={() => imageMouseEnter()}
              onMouseLeave={() => imageMouseLeave()}
            >
              <Image
                altText={currentImage.altText}
                className={styles.image}
                ref={mobileThumbnailRef}
                sizes={currentImage.sizes}
              />
            </div>
          </Hyperlink>
        )}
      </div>
    </Transition>
  );
};

ContentHubArticle.propTypes = {
  articleRef: PropTypes.object,
  category: PropTypes.string,
  className: PropTypes.string,
  dataTestRef: PropTypes.string.isRequired,
  horizontalThumbnail: PropTypes.object,
  id: PropTypes.string,
  isHorizontal: PropTypes.bool,
  isInFirstGroup: PropTypes.bool,
  isMenuItem: PropTypes.bool,
  isReadMore: PropTypes.bool,
  longTitle: PropTypes.string,
  onClick: PropTypes.func,
  readingTime: PropTypes.string,
  uri: PropTypes.string,
  verticalThumbnail: PropTypes.object,
};

ContentHubArticle.defaultProps = {
  articleRef: undefined,
  category: undefined,
  className: undefined,
  dataTestRef: undefined,
  horizontalThumbnail: undefined,
  id: undefined,
  isHorizontal: undefined,
  isInFirstGroup: false,
  isMenuItem: false,
  isReadMore: false,
  longTitle: undefined,
  onClick: undefined,
  readingTime: undefined,
  uri: undefined,
  verticalThumbnail: undefined,
};

export { ContentHubArticle };
