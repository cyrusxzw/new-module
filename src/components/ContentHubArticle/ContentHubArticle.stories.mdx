import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks';
import * as knobs from '@storybook/addon-knobs';
import { StorybookGridOverlay } from '~.storybook/components';
import { ContentHubArticle } from './ContentHubArticle';
import { ContentHubArticleFixture } from './ContentHubArticle.fixture';

<Meta title="Components/ContentHubArticle" component={ContentHubArticle} />

# ContentHubArticle

<Canvas>
  <Story name="Base component">
    {(() => {
      const category = knobs.select(
        'category',
        ['Culture', 'Product', 'The Weekly', 'Aesop Spotify'],
        'Culture',
        'Content',
      );
      const isHorizontal = knobs.select(
        'isHorizontal',
        [true, false],
        true,
        'Content',
      );
      const isReadMore = knobs.select(
        'isReadMore',
        [true, false],
        false,
        'Content',
      );
      const isMenuItem = knobs.select(
        'isMenuItem',
        [true, false],
        false,
        'Content',
      );
      const longTitle = knobs.text(
        'title',
        ContentHubArticleFixture.longTitle,
        'Content',
      );
      const readingTime = knobs.text(
        'readingTime',
        ContentHubArticleFixture.readingTime,
        'Content',
      );
      const implementation = {
        gridOverlay: {
          isActive: knobs.boolean('Show grid overlay', false, 'Implementation'),
          hasInvertedColours: knobs.boolean(
            'Invert grid overlay colours',
            false,
            'Implementation',
          ),
        },
      };
      return (
        <>
          <ContentHubArticle
            id={ContentHubArticleFixture.id}
            category={category}
            dataTestRef={ContentHubArticleFixture.dataTestRef}
            isHorizontal={isHorizontal}
            isReadMore={isReadMore}
            isMenuItem={isMenuItem}
            horizontalThumbnail={ContentHubArticleFixture.horizontalThumbnail}
            longTitle={longTitle}
            onClick={ContentHubArticleFixture.onClick}
            readingTime={readingTime}
            uri={ContentHubArticleFixture.uri}
            verticalThumbnail={ContentHubArticleFixture.verticalThumbnail}
          />
          <StorybookGridOverlay {...implementation.gridOverlay} />
        </>
      );
    })()}
  </Story>
</Canvas>

## API

<ArgsTable of={ContentHubArticle} />
