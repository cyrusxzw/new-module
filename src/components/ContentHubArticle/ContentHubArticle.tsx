import React, { useRef } from 'react';
import cx from 'classnames';
import { useOnScreen } from '~/customHooks';
import { Image } from '~/components/Image';
import { Hyperlink } from '~/components/Hyperlink';
import { Transition } from '~/components/Transition';
import type { ContentHubArticleType } from './ContentHubArticle.types';
import styles from './ContentHubArticle.module.css';

const ContentHubArticle: ContentHubArticleType = ({
  articleRef,
  category,
  className,
  dataTestRef,
  horizontalThumbnail,
  id,
  isHorizontal = false, // use horizontal or vertical thumbnail
  isInFirstGroup = false,
  isMenuItem = false, // if it's displayed in Read Menu Navigation
  isReadMore = false, // if it's displayed in ReadMore section
  longTitle,
  onClick,
  readingTime,
  uri,
  verticalThumbnail,
}) => {
  const defaultRef = useRef(null);
  const ref = articleRef || defaultRef;
  const isOnScreen = useOnScreen(ref, 0.5);
  const currentImage = isHorizontal ? horizontalThumbnail : verticalThumbnail;
  const classSet = cx(styles.base, className, {
    [styles.readMoreArticle]: !!isReadMore,
  });
  const categoryClassSet = cx(styles.category, {
    [styles.readMoreCategory]: !!isReadMore,
  });
  const imageClassSet = cx(styles.image);
  const readMoreThumbnailRef = useRef(null);
  const nonMobileImageClassSet = cx(styles.image, styles.nonMobile);
  const nonMobileThumbnailRef = useRef(null);
  const noteClassSet = cx(styles.note, { [styles.readMoreNote]: !!isReadMore });
  const mobileImageClassSet = cx(styles.image, styles.mobile);
  const mobileThumbnailRef = useRef(null);
  const titleClassSet = cx(styles.title, {
    [styles.readMoreTitle]: !!isReadMore,
  });
  const titleRef = useRef(null);

  const imageMouseEnter = () => {
    const titleSpan = titleRef?.current
      .getElementsByClassName(titleClassSet)[0]
      .getElementsByTagName('span')[0];
    titleSpan.className = cx(styles.title, styles.titleActive);
  };

  const imageMouseLeave = () => {
    const titleSpan = titleRef?.current
      ?.getElementsByClassName(titleClassSet)[0]
      ?.getElementsByTagName('span')[0];
    titleSpan.className = '';
  };

  const titleMouseEnter = () => {
    [readMoreThumbnailRef, nonMobileThumbnailRef, mobileThumbnailRef].forEach(
      (imageRef) => {
        if (imageRef?.current)
          imageRef.current.className = cx(styles.imageActive);
      },
    );
  };

  const titleMouseLeave = () => {
    [readMoreThumbnailRef, nonMobileThumbnailRef, mobileThumbnailRef].forEach(
      (imageRef) => {
        if (imageRef?.current)
          imageRef.current.className = cx(styles.imageUnactive);
      },
    );
  };

  return (
    <Transition isActive={isInFirstGroup || isOnScreen} type="fade">
      <div className={classSet} id={id} ref={ref}>
        {isReadMore && !isMenuItem && (
          <Hyperlink
            className={imageClassSet}
            dataTestRef={`${dataTestRef}_READMORE_THUMBNAIL`}
            onClick={onClick}
            title={longTitle}
            url={uri}
          >
            <div
              onMouseEnter={() => imageMouseEnter()}
              onMouseLeave={() => imageMouseLeave()}
            >
              <Image
                altText={currentImage?.altText}
                ref={readMoreThumbnailRef}
                sizes={currentImage?.sizes}
              />
            </div>
          </Hyperlink>
        )}
        {!isMenuItem && <div className={categoryClassSet}>{category}</div>}
        <div
          onMouseEnter={() => titleMouseEnter()}
          onMouseLeave={() => titleMouseLeave()}
          ref={titleRef}
        >
          <Hyperlink
            className={titleClassSet}
            dataTestRef={`${dataTestRef}_TITLE`}
            onClick={onClick}
            title={longTitle}
            url={uri}
          >
            <span>{longTitle}</span>
          </Hyperlink>
        </div>
        {readingTime && (
          <div className={noteClassSet}>
            <span>{readingTime}</span>
          </div>
        )}

        {!isReadMore && !isMenuItem && (
          <Hyperlink
            className={nonMobileImageClassSet}
            dataTestRef={`${dataTestRef}_NON_MOBILE_THUMBNAIL`}
            onClick={onClick}
            title={longTitle}
            url={uri}
          >
            <div
              onMouseEnter={() => imageMouseEnter()}
              onMouseLeave={() => imageMouseLeave()}
            >
              <Image
                altText={currentImage?.altText}
                className={styles.image}
                ref={nonMobileThumbnailRef}
                sizes={currentImage?.sizes}
              />
            </div>
          </Hyperlink>
        )}
        {!isReadMore && !isMenuItem && (
          <Hyperlink
            className={mobileImageClassSet}
            dataTestRef={`${dataTestRef}_MOBILE_THUMBNAIL`}
            onClick={onClick}
            title={longTitle}
            url={uri}
          >
            <div
              onMouseEnter={() => imageMouseEnter()}
              onMouseLeave={() => imageMouseLeave()}
            >
              <Image
                altText={currentImage?.altText}
                className={styles.image}
                ref={mobileThumbnailRef}
                sizes={currentImage?.sizes}
              />
            </div>
          </Hyperlink>
        )}
      </div>
    </Transition>
  );
};

export { ContentHubArticle };
