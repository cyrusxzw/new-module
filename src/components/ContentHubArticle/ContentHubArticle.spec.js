import React from 'react';
import { render } from '@testing-library/react';
import { ContentHubArticle } from './ContentHubArticle';
import { ContentHubArticleFixture } from './ContentHubArticle.fixture';

describe('<ContentHubArticle />', () => {
  it('should render base component correctly', () => {
    const { container } = render(
      <ContentHubArticle
        category={ContentHubArticleFixture.category}
        dataTestRef={ContentHubArticleFixture.dataTestRef}
        horizontalThumbnail={ContentHubArticleFixture.horizontalThumbnail}
        id={ContentHubArticleFixture.id}
        isHorizontal={true}
        isInFirstGroup={true}
        isMenuItem={false}
        isReadMore={false}
        longTitle={ContentHubArticleFixture.longTitle}
        onClick={ContentHubArticleFixture.onClick}
        readingTime={ContentHubArticleFixture.readingTime}
        uri={ContentHubArticleFixture.uri}
        verticalThumbnail={ContentHubArticleFixture.verticalThumbnail}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
