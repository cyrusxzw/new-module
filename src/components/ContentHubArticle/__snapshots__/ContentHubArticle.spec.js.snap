// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContentHubArticle /> should render base component correctly 1`] = `
<div>
  <div
    class="base"
    id="test_id"
  >
    <div
      class="category"
    >
      Product
    </div>
    <div>
      <a
        class="base left dark title"
        data-test-ref="test_id_TITLE"
        href="https://www.aesop.com/au/r/careers"
        target="_self"
        title="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse et ornare mauris, id dictum ante."
      >
        <h1>
          <span>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse et ornare mauris, id dictum ante.
          </span>
        </h1>
      </a>
    </div>
    <div
      class="note"
    >
      <span>
        5 minute read
      </span>
    </div>
    <a
      class="base left dark image nonMobile"
      data-test-ref="test_id_NON_MOBILE_THUMBNAIL"
      href="https://www.aesop.com/au/r/careers"
      target="_self"
      title="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse et ornare mauris, id dictum ante."
    >
      <div>
        <picture
          class="base picture dark image"
        >
          <source
            media="(min-width: 1025px)"
            srcset="./assets/images/ContentHubArticle/horizontal_hub.jpg"
          />
          <source
            media="(min-width: 640px)"
            srcset="./assets/images/ContentHubArticle/horizontal_hub.jpg"
          />
          <source
            media="(min-width: 0px)"
            srcset="./assets/images/ContentHubArticle/horizontal_hub.jpg"
          />
          <img
            alt="horizontal_hub"
            loading="lazy"
          />
        </picture>
      </div>
    </a>
    <a
      class="base left dark image mobile"
      data-test-ref="test_id_MOBILE_THUMBNAIL"
      href="https://www.aesop.com/au/r/careers"
      target="_self"
      title="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse et ornare mauris, id dictum ante."
    >
      <div>
        <picture
          class="base picture dark image"
        >
          <source
            media="(min-width: 1025px)"
            srcset="./assets/images/ContentHubArticle/horizontal_hub.jpg"
          />
          <source
            media="(min-width: 640px)"
            srcset="./assets/images/ContentHubArticle/horizontal_hub.jpg"
          />
          <source
            media="(min-width: 0px)"
            srcset="./assets/images/ContentHubArticle/horizontal_hub.jpg"
          />
          <img
            alt="horizontal_hub"
            loading="lazy"
          />
        </picture>
      </div>
    </a>
  </div>
</div>
`;
